subsys/net/ip/ipv4.c

```diff
@@ -133,6 +134,18 @@ enum net_verdict net_ipv4_process_pkt(struct net_pkt *pkt)
 	net_pkt_set_ip_hdr_len(pkt, sizeof(struct net_ipv4_hdr));
 	net_pkt_set_ipv4_ttl(pkt, NET_IPV4_HDR(pkt)->ttl);
 
+#ifdef CONFIG_NET_TCP2
+	if (hdr->proto == IPPROTO_TCP) {
+		tcp_input(pkt);
+		goto drop;
```

Comment:
I understand that "goto drop" is a quick hack, but why really update those drop stats which may be again a confusion point?